src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -109,6 +109,11 @@
       (rule-match? x context) (get x "anonymize")
       :else (recur (rest xs)))))
 
+(defn anonymize-text
+  "This is for anonymizing text data where we care only about size, not content."
+  [text]
+  (str/join "" (repeat (count text) "?")))
+
```

Comment:
Suspect this may be a lot more efficient: https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/StringUtils.html#repeat%28char,%20int%29  Or could probably use java.util.Arrays fill (though requiring (local) mutation).